src/base/rss/rssfeed.cpp

```diff
@@ -399,6 +399,32 @@ void Feed::downloadArticleTorrentIfMatching(const ArticlePtr &article)
     BitTorrent::Session::instance()->addTorrent(torrentUrl, params);
 }
 
+// find all occurrences of <placeholder> in original string, and replace them with value of "placeholder" from article
+// example: savePathTpl = "c:\downloads\<attr_name> - <description>\something\<title>\<not_found_attr>"
+//          article = <item>
+//                      <title>some title</title>
+//                      <link>some link</link>
+//                      <guid isPermaLink="false">some-guid</guid>
+//                      <pubDate>Thu, 02 Jun 2016 03:50:04 +0000</pubDate>
+//                      <description>some description</description>
+//                      <attr_name>some attr value</attr_name>
+//                    </item>
+//          result = "c:\downloads\some attr value - some description\something\some title\not_found_attr"
+QString Feed::generateSavePath(const QString &savePathTpl, const ArticlePtr &article)
+{
+    QRegExp rx("<(.+)>");
```

Comment:
Why such unusual syntax (<>)? Why don't you use something familiar? For instance:
```
 savePathTpl = /some/path/$ATTR{attr_name}/another/path/$ATTR{another_attr_name:or_default_value}
```
or
```
 savePathTpl = /some/path/${attr_name}/another/path/${another_attr_name:or_default_value}
```
or
```
 savePathTpl = /some/path/%{attr_name}/another/path/%{another_attr_name:or_default_value}
```